### PR TITLE
Some modification for CIKMCUP

### PR DESCRIPTION
**Pull request by DavdGao**
- Add the dataset `CIKMCUP` and some baselines; 
- Add new metric `imp_ratio`; 
- Fix some bugs for graph regression tasks;

### CODE REVIEW

File: federatedscope/core/monitors/metric_calculator.py [hunk 1]
```diff
@@ -64,22 +64,27 @@ def _check_and_parse(self, ctx):
...
-        # check shape and type
-        if not isinstance(y_true, np.ndarray):
-            raise RuntimeError('Type not support!')
-        if not y_true.shape == y_pred.shape:
-            raise RuntimeError('Shape not match!')
-        if not y_true.ndim == 2:
-            raise RuntimeError(
-                'y_true must be 2-dim arrray, {}-dim given'.format(
-                    y_true.ndim))
+        if 'regression' in ctx.cfg.model.task.lower():
```
Comment (rayrayraykk) [inline federatedscope/core/monitors/metric_calculator.py:67]: All functions `eval_xxx` like `eval_rmse` supports multi-task regression, and should we use `expand_dim` when the task is not multi-task regression?

File: federatedscope/core/monitors/metric_calculator.py [hunk 2]
```diff
@@ -162,18 +167,22 @@ def eval_roc_auc(y_true, y_prob, **kwargs):
...
         # ignore nan values
         is_labeled = y_true[:, i] == y_true[:, i]
         rmse_list.append(
-            np.sqrt(((y_true[is_labeled] - y_pred[is_labeled])**2).mean()))
+            np.sqrt(((y_true[is_labeled] - y_prob[is_labeled]) ** 2).mean()))
 
     return sum(rmse_list) / len(rmse_list)
 
 
+def eval_mse(y_true, y_prob, **kwargs):
```
Comment (rayrayraykk) [inline federatedscope/core/monitors/metric_calculator.py:182]: Multi-task regression?

**Comment by joneswong** [comment]
@wanghh7  could you help us review this pr?

File: federatedscope/core/monitors/metric_calculator.py [hunk 3]
```diff
@@ -64,22 +64,27 @@ def _check_and_parse(self, ctx):
         if torch is not None and isinstance(y_prob, torch.Tensor):
```
Comment (joneswong) [inline federatedscope/core/monitors/metric_calculator.py:64]: please double check what it is. prob, log prob, or logit?

Comment (joneswong) [inline federatedscope/core/monitors/metric_calculator.py:64]: please double check what it is. prob, log prob, or logit?

File: federatedscope/core/monitors/metric_calculator.py [hunk 4]
```diff
@@ -162,18 +167,22 @@ def eval_roc_auc(y_true, y_prob, **kwargs):
     return sum(rocauc_list) / len(rocauc_list)
 
 
-def eval_rmse(y_true, y_pred, **kwargs):
```
Comment (joneswong) [inline federatedscope/core/monitors/metric_calculator.py:165]: we'd better give up this changes. probability is quite confusing in the context of regression that involves mean squired loss.

File: federatedscope/core/monitors/metric_calculator.py [hunk 5]
```diff
@@ -64,22 +64,27 @@ def _check_and_parse(self, ctx):
...
-            raise RuntimeError(
-                'y_true must be 2-dim arrray, {}-dim given'.format(
-                    y_true.ndim))
+        if 'regression' in ctx.cfg.model.task.lower():
+            y_pred = None
+        else:
+            # classification task
+            if y_true.ndim == 1:
+                y_true = np.expand_dims(y_true, axis=-1)
+            if y_prob.ndim == 2:
```
Comment (joneswong) [inline federatedscope/core/monitors/metric_calculator.py:73]: I suggest to remove this augmentation and remove the traversal along with this expanded dimension in `eval_rmse()`

Comment (joneswong) [reply federatedscope/core/monitors/metric_calculator.py:73]: We defer the (completed) support for multi-task setting to the next version. Currently, we'd better focus on single&multi-dimensional regression and single label binary/multi-class classification.

Comment (DavdGao) [reply federatedscope/core/monitors/metric_calculator.py:84]: For now, we add some comment to explain the usage of `y_true`, `y_prob`, and `y_pred`

File: federatedscope/gfl/model/model_builder.py [hunk 6]
```diff
@@ -71,7 +71,7 @@ def get_gnn(model_config, local_data):
                              max_depth=model_config.layer,
                              dropout=model_config.dropout,
                              gnn=model_config.type)
-    elif model_config.task == 'graph':
+    elif model_config.task.startswith('graph'):
```
Comment (rayrayraykk) [inline federatedscope/gfl/model/model_builder.py:74]: We'd better modify `node` and `link` too.

File: federatedscope/gfl/dataset/__init__.py [hunk 7]
```diff
@@ -7,4 +7,4 @@
 from federatedscope.gfl.dataset.kg import KG
 from federatedscope.gfl.dataset.cSBM_dataset import dataset_ContextualSBM
 
-__all__ = ['RecSys', 'DBLPNew', 'KG', 'dataset_ContextualSBM']
+__all__ = ['RecSys', 'DBLPNew', 'KG', 'dataset_ContextualSBM', 'CIKMCupDataset']
```
Comment (joneswong) [inline federatedscope/gfl/dataset/__init__.py:10]: I cannot figure out how to access `CIKMCupDataset`

File: federatedscope/gfl/dataset/preprocess/test.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
+from torch.utils.data import Dataset, DataLoader
+import numpy as np
+
+class MyData(Dataset):
```
Comment (joneswong) [inline federatedscope/gfl/dataset/preprocess/test.py:4]: what is the purpose of this class?

File: federatedscope/gfl/trainer/graphtrainer.py [hunk 9]
```diff
@@ -11,7 +11,11 @@ class GraphMiniBatchTrainer(GeneralTorchTrainer):
     def _hook_on_batch_forward(self, ctx):
         batch = ctx.data_batch.to(ctx.device)
         pred = ctx.model(batch)
-        label = batch.y.squeeze(-1).long()
+        # TODO: deal with the type of data within the dataloader or dataset
+        if 'regression' in ctx.cfg.model.task.lower():
+            label = batch.y
+        else:
+            label = batch.y.squeeze(-1).long()
```
Comment (joneswong) [inline federatedscope/gfl/trainer/graphtrainer.py:18]: this is necessary as some datasets have `batch.y.shape` to be (batch_size, 1), right?

**Review by joneswong — DISMISSED** [review]
LGTM

File: federatedscope/gfl/dataset/__init__.py (hunk 7 above)
Comment (DavdGao) [reply federatedscope/gfl/dataset/__init__.py]: modifed accordingly

File: federatedscope/gfl/dataset/preprocess/test.py (hunk 8 above)
Comment (DavdGao) [reply federatedscope/gfl/dataset/preprocess/test.py:4]: Remove in the new commit

File: federatedscope/gfl/trainer/graphtrainer.py (hunk 9 above)
Comment (DavdGao) [reply federatedscope/gfl/trainer/graphtrainer.py:18]: @rayrayraykk , I'm not sure if it's appropriate to remove `squeeze(-1)`

File: federatedscope/gfl/model/model_builder.py (hunk 6 above)
Comment (DavdGao) [reply federatedscope/gfl/model/model_builder.py:74]: Modified accordingly

**Review by joneswong — DISMISSED** [review]
approved

File: federatedscope/gfl/baseline/isolated_gin_minibatch_on_cikmcup.yaml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    type: SGD
+trainer:
+  type: graphminibatch_trainer
+eval:
+  freq: 5
+  metrics: ['imp_ratio']
+  report: ['avg']
+  best_res_update_round_wise_key: val_loss
+  count_flops: False
+  base: 0.
```
Comment (rayrayraykk) [inline federatedscope/gfl/baseline/isolated_gin_minibatch_on_cikmcup.yaml:37]: Please carefully set this for imp_ratio = `(res-base)/base`

File: federatedscope/gfl/baseline/fedavg_gin_minibatch_on_cikmcup.yaml [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+federate:
+  mode: 'standalone'
+  make_global_eval: False
+  total_round_num: 100
+  share_local_model: False
+  sample_client_num: 16
+data:
+  root: data/
+  type: cikmcup
+  splitter: ooxx
```
Comment (rayrayraykk) [inline federatedscope/gfl/baseline/fedavg_gin_minibatch_on_cikmcup.yaml:16]: We'd better set `ooxx` to `none` or `nan' to be more forma.

File: federatedscope/gfl/baseline/isolated_gin_minibatch_on_cikmcup.yaml [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+  mode: standalone
+  method: local
+  make_global_eval: False
+  total_round_num: 10
+  share_local_model: False
+data:
+  batch_size: 64
+  root: data/
+  type: cikmcup
+  splitter: ooxx
```
Comment (rayrayraykk) [inline federatedscope/gfl/baseline/isolated_gin_minibatch_on_cikmcup.yaml:17]: We'd better set `ooxx` to `none` or `nan' to be more forma.

**Review by rayrayraykk — APPROVED** [review]
After modification, it looks good to me.

**Review by joneswong — APPROVED** [review]
approved.
